scripts/test-harness-contracts.ps1

```
@@ -944,6 +944,82 @@ try {
+        Run-Native { node $syncTest } 'sync-harness: contract-shape behavior (orphan #17, duplicate #21, healthy paths)'
+    } finally {
+        Remove-Item -LiteralPath $syncTest -Force -ErrorAction SilentlyContinue
+    }
+
+    # #19 — the check must REFUSE to skip the entrypoints; --internal-staging is the dist exception.
+    $skipOut = (& node (Join-Path $root 'scripts/sync-harness.mjs') --check --skip-entrypoints 2>&1 | Out-String)
+    Check ($LASTEXITCODE -ne 0) 'sync --check --skip-entrypoints fails instead of reporting a green that verified nothing' "exit $LASTEXITCODE"
+    Check ($skipOut -match 'cannot be combined with --skip-entrypoints') 'sync names why the check cannot skip the entrypoints'
+    Check-Match 'scripts/sync-harness.mjs' "anchorEntrypoints \? \[\] : \['--internal-staging'\]" 'dist staging uses the internal flag, not the user-facing skip'
+    Check-Match 'scripts/sync-harness.mjs' 'export \{ scanContractRegions, extractContract, upsertContract \}' 'sync exports the contract-shape helpers so the suite can exercise them'
+    Check-Match 'scripts/sync-harness.mjs' 'importing the module must have no side effects' 'sync only runs the CLI when invoked directly'
```

PSScriptAnalyzer (1.25.0)

[info] Cmdlet 'Check-Match' has positional parameter. Please use named parameters instead of positional parameters when calling a command.

(PSAvoidUsingPositionalParameters)